Compilation in a cluster using a new mechanism model

I am facing a compilation issue within a cluster, specifically related to the creation of the executable file.

I created a new mechanism within the 'Mechanism' section of PelePhysics. This mechanism works well on my local desktop for simplified cases, allowing me to obtain the .exe file and run computations successfully.

However, when I attempt to compile it on the cluster, I am unable to generate the .exe file, and I receive the following error message: /PeleLMeX-1/Submodules/PelePhysics/ThirdParty/INSTALL/gnu/lib/libsundials_cvode.so: undefined reference to `pow@GLIBC_2.29'
collect2: error: ld returned 1 exit status
make: *** [/PeleLMeX-1/Submodules/amrex/Tools/GNUMake/Make.rules:56: PeleLMeX2d.gnu.ex] Error 1

Interestingly, when I switch to a different mechanism in the GNUmakefile, the compilation process continues until it reaches the linking stage and generates the .exe file.

Do you have any suggestions on how to avoid such issues using the new mechanism, particularly when working with a cluster?

--- a/.github/workflows/dependencies/dependencies_dpcpp.sh
+++ b/.github/workflows/dependencies/dependencies_dpcpp.sh
@@ -9,7 +9,7 @@ set -eu -o pipefail
 
 # Ref.: https://github.com/rscohn2/oneapi-ci
 # intel-basekit intel-hpckit are too large in size
-wget -q -O - https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB \
+wget -q -O - https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB \
   | sudo apt-key add -
 echo "deb https://apt.repos.intel.com/oneapi all main" \
   | sudo tee /etc/apt/sources.list.d/oneAPI.list
@@ -18,7 +18,7 @@ sudo apt-get update
 
 sudo apt-get install -y --no-install-recommends \
     build-essential \
-    intel-oneapi-dpcpp-cpp-compiler intel-oneapi-mkl-devel \
+    intel-oneapi-compiler-dpcpp-cpp intel-oneapi-mkl-devel \
     g++             \
     libopenmpi-dev  \
     openmpi-bin

--- a/README.md
+++ b/README.md
@@ -75,7 +75,7 @@ cd PeleLMeX/Exec/RegTest/HotBubble
 
 If this is a clean install, you will need to make the third party libraries with: `make TPL` (note: if on macOS, you might need to specify `COMP=llvm` in the `make` statements).
 
-Finally, make with: `make -j`, or if on macOS: `make -j COMP=llvm`. To clean the installation, use either `make clean` or `make realclean`.
+Finally, make with: `make -j`, or if on macOS: `make -j COMP=llvm`. To clean the installation, use either `make clean` or `make realclean`. If running into compile errors after changing compile time options in PeleLMeX (e.g., the chemical mechanism), the first thing to try is to clean your build by running `make TPLrealclean && make realclean`, then try to rebuild the third party libraries and PeleLMeX with `make TPL && make -j`.
 
 When using CMake (cmake version >= 3.23 is required), first configure CMake directly in *PeleLMeX* root folder:
 

